crates/bevy_ecs/src/component/mod.rs

```diff
@@ -0,0 +1,288 @@
+mod type_info;
+
+pub use type_info::*;
+
+use crate::storage::SparseSetIndex;
+use bitflags::bitflags;
+use std::{
+    alloc::Layout,
+    any::{Any, TypeId},
+    collections::hash_map::Entry,
+};
+use thiserror::Error;
+
+pub trait Component: Send + Sync + 'static {}
+impl<T: Send + Sync + 'static> Component for T {}
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub enum StorageType {
+    Table,
+    SparseSet,
+}
+
+impl Default for StorageType {
+    fn default() -> Self {
+        StorageType::Table
+    }
+}
+
+#[derive(Debug)]
+pub struct ComponentInfo {
+    name: String,
+    id: ComponentId,
+    type_id: Option<TypeId>,
+    // SAFETY: this must remain private. it should only be set to "true" if this component is actually Send + Sync
```

Comment:
```suggestion
    // SAFETY: this must remain private. it must only be set to "true" if this component is actually Send + Sync
```
Must is stronger than should.

Reply:
There's also a typo here; it's missing a capital.

Reply:
There are capitals missing in pretty much every SAFETY comment.